src/openzaak/components/documenten/models.py

```diff
@@ -776,25 +816,37 @@ class Meta:
             # mutual exclusive check on zaak fk, besluit fk or object url
             models.CheckConstraint(
                 check=(
-                    Q(_object_url="", _zaak__isnull=False, _besluit__isnull=True)
-                    | Q(_object_url="", _zaak__isnull=True, _besluit__isnull=False)
-                    | Q(~Q(_object_url=""), _zaak__isnull=True, _besluit__isnull=True)
+                    Q(
+                        _object_base_url__isnull=True,
+                        _zaak__isnull=False,
+                        _besluit__isnull=True,
+                    )
+                    | Q(
+                        _object_base_url__isnull=True,
+                        _zaak__isnull=True,
+                        _besluit__isnull=False,
+                    )
+                    | Q(
+                        _object_base_url__isnull=False,
+                        _zaak__isnull=True,
+                        _besluit__isnull=True,
+                    )
```

Comment:
I'd also expect an additional check constraint that a relative part must be filled out when the service FK is not null (separate check, don't add it to this one)

Reply:
the check constraint is added to zgw_consumers.ServiceUrlField (PR - https://github.com/maykinmedia/zgw-consumers/pull/56)